packages/core/src/providers/azure-ad.ts

```diff
@@ -135,6 +136,26 @@ export default function AzureAD<P extends AzureADProfile>(
       params: {
         scope: "openid profile email User.Read",
       },
+      async conform(response) {
+        // Microsoft being special and non-compliant #9635
+        if (response.status !== 200) return response
+
+        let json = await response.json()
+        json.issuer = json.issuer.replace("{tenantid}", tenantId)
+        return new Response(JSON.stringify(json), response)
+      },
+      async serverConform(authServer, codeGrantResponse) {
+        let tenantId: string = "common";
+        if (codeGrantResponse.status === 200) {
+          const { id_token } = await codeGrantResponse.clone().json()
+            ; ({ tid: tenantId } = jose.decodeJwt(id_token))
+        }
+
+        return {
+          ...authServer,
+          issuer: authServer.issuer.replace("common", tenantId),
```

Comment:
This isn't right. Need to detect the tenant that's being used. Probably just replace the provider's tenant ID with whatever we expect from the JWT.